apps/extension/src/web-worker.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Action,
+  TransactionPlan,
+  WitnessData,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1alpha1/transaction_pb';
+import { JsonValue } from '@bufbuild/protobuf/dist/esm';
+
+interface WorkerMessagePayload {
+  transactionPlan: JsonValue;
+  witness: JsonValue;
+  fullViewingKey: string;
+  actionId: number;
+}
+
+const workerListener = ({ data }: { data: WorkerMessagePayload }) => {
+  const {
+    transactionPlan: transactionPlanJson,
+    witness: witnessJson,
+    fullViewingKey,
+    actionId,
+  } = data;
+
+  // Deserialize payload
+  const transactionPlan = TransactionPlan.fromJson(transactionPlanJson);
+  const witness = WitnessData.fromJson(witnessJson);
+
+  void executeWorker(transactionPlan, witness, fullViewingKey, actionId).then(action => {
+    self.postMessage(action);
+  });
```

Comment:
I thought about this. @grod220 mentioned we should be using `fromJson`. Can we keep it as a `JsonValue` for the entire build process?

Reply:
currently the behavior of the wasm-ts wrapper is to validate zod schema and deserialize with fromJson, and this works.

but, in the wasm-task worker, we're immediately re-serializing and posting it to the parent offscreen, which also immediately needs it serialized to post it back to the caller of the offscreen work.

i'm still of the opinion that we're doing excessive ser/de, but let's just go for consistency, and do it like the other wrapper methods. this can be a separate issue.